src/hooks/useTableTools/useTableTools.js

```diff
@@ -63,18 +62,10 @@ const useTableTools = (
   const { columns, columnManagerAction, columnManagerModalProps } =
     useColumnManager(options);
 
-  // TODO extract to separate hook
-  const { toolbarProps: toolbarActionsProps } = useMemo(
-    () =>
-      toToolbarActions({
-        ...options,
-        firstAction: dedicatedAction,
-        actions: [
-          ...(options?.actions || []),
-          ...((columnManagerAction && [columnManagerAction]) || []),
-        ],
-      }),
-    [columnManagerAction, options, dedicatedAction],
+  const { toolbarProps: toolbarActionsProps } = useToolbarActions(
+    options,
+    dedicatedAction,
```

Comment:
The `dedicatedAction` is also within the options, which means we don't need pass it in separately, but can extract it from the options within the `useToolbarActions` hook, similar to how it is shown above at around line 46/47